packages/block-editor/src/store/selectors.js

```diff
@@ -1862,6 +1863,14 @@ const canIncludeBlockTypeInInserter = ( state, blockType, rootClientId ) => {
 const getItemFromVariation = ( state, item ) => ( variation ) => {
 	const variationId = `${ item.id }/${ variation.name }`;
 	const { time, count = 0 } = getInsertUsage( state, variationId ) || {};
+	const supportsAlias = variation?.supports?.alias ?? false;
+	const initialAttributes = {
+		...item.initialAttributes,
+		...variation.attributes,
+		...( supportsAlias && {
+			className: getBlockDefaultClassName( variation.name ),
```

Comment:
Maybe it would be better to move this to what generates the markup rather than a pre-set attribute?

Reply:
Potential problem with this is block validation

Reply:
Additional custom class names are ignored during validations, so that should be good. That's also a design question whether we want to have two class names printed:
- computed from the original block type
- computer from the alias for the block variation